ai-compile/vscode-extension/src/extension.ts

```diff
@@ -0,0 +1,170 @@
+import * as vscode from 'vscode';
+import { compile, getFix, LocalStorage } from './util';
+import { CodelensProvider } from './code-lens/codeLensProvider';
+
+import * as t from './types';
+import * as c from './constants';
+
+let disposables: vscode.Disposable[] = [];
+let eventDisposables: vscode.Disposable[] = [];
+
+// file-wise history store
+export let storageManager: LocalStorage;
+
+async function compileAndGetFixHelper(
+): Promise<t.Fix> {
```

Comment:
This seems odd to me. We already have async/await, which uses Promise under the hood. So something seems off here.

Reply:
I see now... you've just explicitly annotated the types.